components/function-controller/Makefile

```diff
@@ -31,20 +38,36 @@ test-local:
 
 # Generate manifests e.g. CRD, RBAC etc.
 .PHONY: manifests
-manifests: generate-local
+manifests: controller-gen-local generate-local
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=serverless webhook paths="./..." \
 		object:headerFile=hack/boilerplate.go.txt \
 		output:crd:artifacts:config=config/crd/bases \
 		output:rbac:artifacts:config=config/rbac \
 		output:webhook:artifacts:config=config/webhook
 
-CONTROLLER_GEN := $(shell which controller-gen)
-
 # Generate code
 .PHONY: generate-local
-generate-local:
+generate-local: controller-gen-local
 	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
 
+# find or download controller-gen
+# download controller-gen if necessary
+.PHONY: controller-gen-local
+controller-gen-local:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.2.9 ;\
```

Comment:
Are there any breaking changes between version 0.2.9 and 0.2.4? I found in our `README.md` that we require `controller-gen (v0.2.4+)`. Maybe can we change this to `controller-gen (v0.2.9)`? Or is this not necessary?

Reply:
I can update the readme with the new version but looking at the [changelog of controller-tools](https://github.com/kubernetes-sigs/controller-tools/releases) I don't see any breaking changes from `0.2.4` to `0.2.9`. Used this version as it was the version our current CRDs were manually generated with in recent updates related to Kyma 2.0.